### ISSUE TITLE
Compile only "changed files" from watch command

### ISSUE THREAD

**Issue by flexmotion** [jescalan/roots#714]
Is it possible through the roots build scripts to set a setting that would allow for only changed files to get compiled?

Atm. with < 20 files it takes several seconds to compile any change.

This is primarily changes to ex. .styl, .coffee etc.

**Comment by jescalan** [comment]
Unfortunately it is not, because of dependencies - see #586 for a full explanation.

However, we are working on a next-generation version of roots called [spike](https://github.com/static-dev/spike) that holds on to a full dependency graph and solves this problem. It's in early stages so I wouldn't yet build any important production apps on top of it unless you want to be in close contact with us about bugs and breaking changes. Either way, in our experience so far it's been significantly faster in watch mode, so if you want feel free to keep an eye on it!

**Comment by flexmotion** [comment]
So Roots is getting "put on ice"?

**Comment by jescalan** [comment]
The plan at the moment is once spike is fully stable and released to put Roots on "LTS", basically in maintenance mode. No new features will be added, but bugs will be fixed, dependencies will stay up to date, and everything will continue working as is. Down the road we will decide how long of a period this will be, but it will be at least a year.

Essentially, we have done what we can with Roots, it works well for us and many others, and we use it at my company in production for a large number of sites. But we have run up against limitations that require significant restructures to resolve, and therefore are moving towards a different project in order to not push another roots release that will be so different that migrating to it from an existing roots project would be an obscene amount of work.

**tl;dr** Roots will stay entirely stable and well maintained into the future, but if you want bleeding edge features, we can make that happen with spike.
